components/forecast/WeatherTab.tsx

```diff
@@ -144,16 +153,26 @@ export const WeatherTab: React.FC<WeatherTabProps> = ({zone}) => {
               <Body>Error loading weather stations.</Body>
             </HStack>
           )}
-          {weatherStationStatus === 'success' &&
-            groupedWeatherStations.length > 0 &&
-            groupedWeatherStations.map(([name]) => (
-              <HStack key={name} borderBottomWidth={1} borderColor={colorLookup('light.200')} py={8}>
-                <VStack>
-                  <Body>{name}</Body>
-                  {/* <BodyXSm>stations: {stations.map(s => s.stid).join(',')}</BodyXSm> */}
-                </VStack>
-              </HStack>
-            ))}
+          {weatherStationStatus === 'success' && groupedWeatherStations.length > 0 && (
+            <ActionList
+              actions={groupedWeatherStations.map(([name, stations]) => ({
+                label: name,
+                data: stations,
+                action: () => {
+                  // Nested navigation to the stationDetail page of the Weather Data stack
+                  const dateString = apiDateString(date);
+                  navigation.navigate('Weather Data', {
+                    center_id,
+                    dateString,
```

Comment:
slight change from previous revision - now `center_id` and `dateString` have to be passed as params for both the `Weather Data` screen, and the `stationDetail` screen. kinda weird, but livable I think.